File: week-03/day-1/Workshop 17.4/src/CopyFile.java
import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;


public class CopyFile {
    public static void main(String[] args) {
        // Write a `copy` function that copies the contents of a file into another file.
        // The function should take two filenames (strings) as parameters:
        //  - path and name of the source file
        //  - path and name of the target file
        // It should return a boolean that shows whether the copy was successful or not.

        try { copy("resources/greetings.txt", "resources/my-file.txt" );
        } catch (IOException e) {
            System.out.println( "Unable to do.");
        }

    }
   public static void copy (String filename1, String filename2) throws IOException {
       File file1 = new File(filename1);
       File file2 = new File(filename2);
       Path filePath1 = Paths.get(filename1);
       Path filePath2 = Paths.get(filename2);

       Files.copy(filePath1, filePath2);
   }

}

File: week-07/day-2/Workshop 16.5.2023/copy_file.py
# Write a function that copies the contents of a file into another file
# It should take two filenames as parameters
# It should return a boolean that shows if the copy was successful

def copy(filename, filename2):
    try:
        file_to_copy = open(filename, 'r')
        file_to_paste = open(filename2, 'w')
        for line in file_to_copy:
            file_to_paste.write(line)
        return True
    except IOError:
        return False


copy('apple_text.txt', 'new_apple_text.txt')